components/script/dom/filereader.rs

```diff
@@ -176,14 +191,26 @@ impl FileReader {
         fr.change_ready_state(FileReaderReadyState::Done);
         // Step 8.2
 
-        let output = match data.function {
+        let output: StringOrObject = match data.function {
             FileReaderFunction::ReadAsDataUrl =>
                 FileReader::perform_readasdataurl(data, &blob_contents),
             FileReaderFunction::ReadAsText =>
                 FileReader::perform_readastext(data, &blob_contents),
+            FileReaderFunction::ReadAsArrayBuffer =>
+                FileReader::perform_readasarraybuffer(fr.global().r().get_cx(), data, &blob_contents),
         };
-
-        *fr.result.borrow_mut() = Some(output);
+        *fr.result.borrow_mut() = match output {
+            StringOrObject::String(ref string) => Some(FileReaderResult::String_(string.clone())),
+            StringOrObject::Object(obj) => Some(FileReaderResult::ArrayBuffer_(Heap::default())),
+        };
+        if let Some(FileReaderResult::ArrayBuffer_(ref heap)) = *fr.result.borrow() {
+            if let StringOrObject::Object(obj) = output {
+                unsafe {
+                    let tempr = heap.get_unsafe();
```

Comment:
I had tried that, but I throws a `cannot borrow immutable borrowed content *heap as mutable` error. Trying that with a mutable reference throws a  `cannot borrow anonymous inner field as mutable` error.